protocols/identify/src/behaviour.rs

```diff
@@ -291,11 +291,15 @@ impl NetworkBehaviour for Behaviour {
                 self.discovered_peers
                     .put(peer_id, info.listen_addrs.iter().cloned());
 
-                let observed = info.observed_addr.clone();
+                let observed_addr = info.observed_addr.clone();
                 self.events
                     .push_back(ToSwarm::GenerateEvent(Event::Received { peer_id, info }));
                 self.events
-                    .push_back(ToSwarm::NewExternalAddrCandidate(observed));
+                    .push_back(ToSwarm::NewExternalAddrCandidate(observed_addr.clone()));
+                self.events.push_back(ToSwarm::NewExternalAddrOfPeer {
+                    peer_id,
+                    addr: observed_addr,
```

Comment:
This is the address that the other node observed of _us_. What we want to emit here are the self-advertised addresses of the other peer contained in `info`!

Reply:
I'd like to see a test for this:

- Connect two swarms
- One swarm adds an external address
- 2nd swarm receives this via identify push and emits this event
- assert that 2nd swarm receives this event

Reply:
> What we want to emit here are the self-advertised addresses of the other peer contained in info!

Hmm, so in that case should the `NewExternalAddrOfPeer` have Vec of `Multiaddr`s? Should we handle a case where we get multiple new external addresses in `Info`?

Reply:
> > What we want to emit here are the self-advertised addresses of the other peer contained in info!
> 
> Hmm, so in that case should the `NewExternalAddrOfPeer` have Vec of `Multiaddr`s? Should we handle a case where we get multiple new external addresses in `Info`?

I don't think there is a need for a `Vec`, we can just emit the same event multiple times right?

Reply:
I've added related changes - struggling a bit with a smoke test for swarms atm

Reply:
Are you joining the [maintainers call](https://github.com/libp2p/rust-libp2p/discussions/4606) tomorrow? Happy to talk about it here.

Reply:
Hi @thomaseizinger :wave: 
I'll join the next maintainers call to talk about these changes

Reply:
Awesome, looking forward to it!